src/testdir/test_autocmd.vim

```diff
@@ -2096,6 +2096,21 @@ func Test_Cmdline()
   au! CmdlineEnter
   au! CmdlineLeave
   let &shellslash = save_shellslash
+
+  au! CursorMovedC : let g:pos = getcmdpos()
+  let g:pos = 0
+  call feedkeys(":hello\<Left>\<ESC>", 'xt')
+  call assert_equal(5, g:pos)
+  call feedkeys(":12345678\<C-R>=setcmdpos(3)\<CR>\<ESC>", 'xt')
+  call assert_equal(3, g:pos)
+  au! CursorMovedC
+
+  " CursorMovedC changes the cursor position.
+  au! CursorMovedC : let g:pos = getcmdpos() | call setcmdpos(getcmdpos()-1)
```

Comment:
It seems that this test doesn't really make sense, because `setcmdpos()` doesn't actually work when using an autocommand:
```
		Only works while editing the command line, thus you must use
		|c_CTRL-\_e|, |c_CTRL-R_=| or |c_CTRL-R_CTRL-R| with '='.  For
		|c_CTRL-\_e| and |c_CTRL-R_CTRL-R| with '=' the position is
		set after the command line is set to the expression.  For
		|c_CTRL-R_=| it is set after evaluating the expression but
		before inserting the resulting text.
```

Reply:
Hm....  If so, why it is nesting?

https://github.com/vim/vim/pull/15040#issuecomment-2177285810

Reply:
Because you are triggering `CursorMovedC` even if cursor didn't move.

Reply:
It seems that the initial value of `save_cmdspos` is also wrong, as it is using the position from the previous cmdline. I'll fix these problems in #15064.

Reply:
Oh...